base/src/main/java/org/aya/concrete/visitor/PatternFolder.java

```diff
@@ -30,15 +31,26 @@
     return foldVar(acc, var, pos, type);
   }
 
+  default @NotNull R fold(@NotNull R acc, @NotNull Arg<Pattern> pat) {
+    return fold(acc, pat.term());
+  }
+
   @MustBeInvokedByOverriders
   default @NotNull R fold(@NotNull R acc, @NotNull Pattern pat) {
     return switch (pat) {
+      case Pattern.List list -> list.elements().foldLeft(acc, this::fold);
+      case Pattern.Tuple tuple -> tuple.patterns().foldLeft(acc, this::fold);
+      case Pattern.BinOpSeq seq -> seq.seq().foldLeft(acc, this::fold);
```

Comment:
Traverse nested patterns are done in 

https://github.com/aya-prover/aya-dev/blob/723d709514b067719746c0a98f92c4dde1feb3fc/base/src/main/java/org/aya/concrete/visitor/StmtFolder.java#L61-L80

So it is not needed here.

Reply:
Same for `Expr` and `Stmt`